File: backend/schemas/journal.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
import uuid
from typing import List

class JournalEntryCreate(BaseModel):
    content: str

class JournalEntryUpdate(BaseModel):
    content: str

class JournalEntryResponse(BaseModel):
    id: uuid.UUID
    case_id: uuid.UUID
    content: str
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

class JournalEntryListResponse(BaseModel):
    entries: List[JournalEntryResponse]

File: app-frontend/medha-app/src/types/journal.ts
export interface JournalEntryCreate {
  content: string;
}

export interface JournalEntryUpdate {
  content: string;
}

export interface JournalEntryResponse {
  id: string;
  case_id: string;
  content: string;
  created_at: string;
  updated_at: string;
}

export interface JournalEntryListResponse {
  entries: JournalEntryResponse[];
}
